packages/core/src/contrib/ClosestPoint.test.ts

```diff
@@ -0,0 +1,80 @@
+import { genCode, secondaryGraph } from "engine/Autodiff";
+import { makeCircle } from "shapes/Circle";
+import { makeRectangle } from "shapes/Rectangle";
+import { Context, makeCanvas, simpleContext } from "shapes/Samplers";
+import { Shape } from "types/shapes";
+import { black, floatV, vectorV } from "utils/Util";
+import { compDict } from "./Functions";
+
+const canvas = makeCanvas(800, 700);
+
+const compareClosestPoint = (
```

Comment:
May be helpful to add a /** */ comment to each function to document its meaning and the meaning of its input / output values.